### ISSUE TITLE
MultipoleProfilePotential: continue analytically outside [r_min, r_max]

### ISSUE THREAD

**Issue by nstarman** [GalacticDynamics/galax#850]
Follow-up to the `MultipoleProfilePotential` port.

## Problem

`MultipoleProfilePotential` splines Phi_lm(r) and rho_lm(r) on a log grid over
`[r_min, r_max]` and evaluates outside that range by continuing the edge cubic
in log r (`interpax.CubicHermiteSpline(..., extrapolate=True)` in
`_src/builtin/multipole_profile/funcs.py`). There is no analytic tail, so the
result degrades rapidly and without warning.

Measured for a Hernquist sphere (`m_tot=1e12 Msun`, `r_s=10 kpc`) expanded with
`r_min=1e-2 kpc`, `r_max=300 kpc`, `n_r=128`, `l_max=0`:

| r (kpc) | r/r_max | Phi | truth | rel. error |
|---|---|---|---|---|
| 350 | 1.17 | -1.2187e-02 | -1.2496e-02 | 0.025 |
| 500 | 1.67 | -2.3711e-03 | -8.8206e-03 | 0.73 |
| 700 | 2.33 | **+1.9654e-02** | -6.3359e-03 | 4.10 |
| 1000 | 3.33 | **+6.5927e-02** | -4.4540e-03 | 15.8 |

The potential changes sign around 700 kpc. Density is worse — roughly 5e5 times
too large at 2*r_max and growing.

This matters more in `galax` than it did upstream. `bfeax` is a research tool
evaluated on its own grid; a `galax` potential gets handed to an ODE integrator
that decides for itself where to go, so an orbit that leaves the grid silently
produces nonsense.

Currently this is documented on the class and both constructors, and pinned by
`test_outer_extrapolation_diverges_from_truth`, but not fixed.

## Proposed fix

Continue analytically instead of extrapolating the spline. Outside the grid the
solution of Laplace's equation is a pure power law per mode:

```
r > r_max:  Phi_lm(r) = Phi_lm(r_max) * (r / r_max)^{-(l+1)}
r < r_min:  Phi_lm(r) = Phi_lm(r_min) * (r / r_min)^{l}
```

`poisson.py` already fits exactly these power laws for its boundary-tail
corrections, so the exponents are on hand. Density should go to its own
power-law continuation (the inner one is already stored as `rho_alpha` /
`rho_amplitude`) or to zero beyond `r_max`, whichever is decided.

## Acceptance

- Relative error outside `[r_min, r_max]` bounded and monotone, with no sign
  change.
- `test_outer_extrapolation_diverges_from_truth` updated deliberately — it
  exists to force exactly that (see the note below).
- Closed-form comparisons extended to sample outside the grid.

## Note

That pinning test currently asserts `error_ratio > 0.01` at r=700, where the
measured ratio is 4.10 — 410x above the threshold. Its docstring claims it will
fail once the analytic tail lands, but at that threshold an implementation that
merely got the error down to a few percent would still pass. Whoever takes this
issue should tighten it (and ideally assert the sign flip, which is the
qualitative signature a real continuation removes).

---

## How the reference implementations handle this

**`bfeax` does not solve it.** Its `spline.spline_eval` docstring says
"Clamps to boundary for out-of-range queries", but the code clamps only the
interval *index* and then evaluates with the true offset:

```python
idx = jnp.searchsorted(x_knots, x_query, side="right") - 1
idx = jnp.clip(idx, 0, x_knots.shape[0] - 2)
dt  = x_query - x_knots[idx]          # NOT clamped
return a[idx] + b[idx]*dt + c[idx]*dt**2 + d[idx]*dt**3
```

That is edge-cubic extrapolation, which is what this port inherited. `bfeax`'s
only asymptotic handling is inside the Poisson *solve* (the boundary
corrections to the two radial integrals), not in evaluation.

**Agama does solve it**, and its scheme is a better target than the simple
power law proposed above. See `PowerLawMultipole` and `initAsympt()` /
`computeExtrapolationCoefs()` in
[`src/potential_multipole.cpp`](https://github.com/GalacticDynamics-Oxford/Agama/blob/master/src/potential_multipole.cpp):

```
Phi(r) = W (r/r1)^v  +  U (r/r1)^s  +  Q (r/r1)^2
```

(and `W (r/r1)^v + U (r/r1)^s ln(r/r1)` in the degenerate case `s == v`),
with

- `v = l` for inward extrapolation (`r < r_min`), `v = -l-1` for outward
  (`r > r_max`) — these are the source-free solid-harmonic solutions, i.e.
  what the simple proposal above already had;
- `s` the asymptotic slope of the *density*, found by root-finding
  (`DensityAsymptoteFinder`) from `Phi` and `dPhi/dr` at the two nearest grid
  nodes;
- `W`, `U`, `Q` solved algebraically from those four boundary constraints.

So the `U (r/r1)^s` term carries the contribution of mass outside (or inside)
the grid, and `Q (r/r1)^2` the uniform-density / external-field piece — the
parts a pure `r^l` / `r^-(l+1)` continuation drops.

Agama also guards the fitted slope: density may not grow faster than `r^-3`
at small `r`, and must fall faster than `r^-2` at large `r` — otherwise the
enclosed mass diverges. Worth copying; this port already computes closely
related boundary slopes in `poisson.py` for its tail corrections, and the
same convergence conditions appear there.

**Implication for the proposal above.** The simple `Phi_lm ∝ r^-(l+1)` /
`r^l` continuation is the right *shape* but is only the `W` term. It will be
markedly better than the current unguarded cubic, and may be enough — but
matching Agama means fitting `s` and carrying `U` and `Q` too. Worth
measuring both against a case with known asymptotics before choosing.

**Comment by nstarman** [comment]
Landed for the potential across #859 (machinery) and #852 (wiring).

**What shipped.** Rather than the pure `r^{-(l+1)}` / `r^l` continuation proposed above, #859 implements Agama's `PowerLawMultipole` form,

    Phi(x) = P1 x^v + B (x^s - x^v)/(s - v) + Q (x^2 - x^v),   x = r/r_boundary

fitted per mode by `asymptotic_coeffs` and evaluated by `eval_log_spline_asympt`. The pure power law is the `B = Q = 0` special case. Writing it this way pins the join to the spline in **value and slope** to round-off — the boundary value is `P1` bit-for-bit and the boundary derivative is `v P1 + B + (2-v) Q`, so `B` is chosen to close it exactly. Agama lets both follow from its algebra and is C1 only to the accuracy of that algebra. Measured here: C1 mismatch 2.06 eps in value, 1.88 eps in slope; Kepler monopole exact to <=2 eps with `U = Q = 0`; the interior is bitwise identical (0.0).

**Result**, same configuration as the table above:

| r/r_max | edge cubic | fitted tail |
|---|---|---|
| 1.17 | 0.0247 | 0.0020 |
| 1.67 | 0.7312 | 0.0098 |
| 2.33 | **4.1020** (sign flip) | 0.0152 |
| 3.33 | **15.8019** (sign flip) | 0.0194 |
| 10.00 | -- | 0.0259 |

The sign flip is gone at every radius — the tail is anchored on the boundary value, so a bound monopole cannot cross zero going outward.

**The NaN gradient at the origin is fixed too**, which this issue also tracked. The inner tail is evaluated on a clamped `log r`, so `Phi(0)` is finite (-0.4530 against a true -0.4499) and `grad(0)` is exactly zero, matching the analytic potential. That was the more dangerous of the two defects, since one NaN poisons an entire vmapped batch of orbits.

**Two things this does not do**, both deliberate:

1. **The density is not continued.** The fitted form is the potential's asymptotic structure, not a density's, so applying it to `rho_lm` would be wrong physics. `_density` outside the grid is still the edge cubic on the residual and can still grow unbounded. This is now the limitation the class docstring records.
2. **Accuracy outside is capped by the spline's end condition.** `fit_log_spline` uses a natural end condition, which caps the continuation 50-100x short of what analytic boundary derivatives give (4.7e-4 at 2 r_max instead of 2.4e-2 for Hernquist). Fixing that would perturb the interior, which the bit-identical requirement forbids. Tracked separately at #858.

Closing once #852 merges. Leaving open until then so the trail stays attached to the work.
